integration_tests/sdk/delete_workflow_test.py

```diff
@@ -46,3 +46,135 @@ def test_delete_workflow_invalid_saved_objects(client):
         client.flow(flow_id)
     finally:
         delete_flow(client, flow_id)
+
+
+# Check the flow with object(s) saved with update_mode=APPEND can only be deleted if in force mode.
+@pytest.mark.publish
+def test_delete_workflow_saved_objects(client):
+    integration = client.integration(name=get_integration_name())
+    name = generate_new_flow_name()
+    table_name = generate_new_flow_name()
+    flow_ids_to_delete = set()
+
+    ###
+
+    table = integration.sql(query=SHORT_SENTIMENT_SQL_QUERY)
+
+    table.save(integration.config(table=table_name, update_mode=LoadUpdateMode.REPLACE))
+
+    flow_ids_to_delete.add(
+        run_flow_test(client, [table], name=name, num_runs=1, delete_flow_after=False).id()
+    )
+
+    ###
+
+    table.save(integration.config(table=table_name, update_mode=LoadUpdateMode.APPEND))
+
+    flow_ids_to_delete.add(
+        run_flow_test(client, [table], name=name, num_runs=2, delete_flow_after=False).id()
+    )
+
+    ###
+
+    try:
+        assert len(flow_ids_to_delete) == 1
+        flow_id = list(flow_ids_to_delete)[0]
+        tables = client.flow(flow_id).list_saved_objects()
+
+        assert table_name in [item.object_name for item in tables[get_integration_name()]]
+
+        # Check table is properly created at the integration.
+        # Need to poll initially in case still writing table.
+        check_table_exists(integration, table_name)
+
+        # Doesn't work if don't force because it is created in append mode.
+        with pytest.raises(InvalidRequestError):
+            client.delete_flow(flow_id, saved_objects_to_delete=tables, force=False)
+
+        # Check table is properly created at the integration.
+        integration.sql(f"SELECT * FROM {table_name}").get()
+
+        client.delete_flow(flow_id, saved_objects_to_delete=tables, force=True)
+
+        # Check flow indeed deleted
+        check_flow_doesnt_exist(client, flow_id)
+        flow_ids_to_delete.remove(flow_id)
+
+        # Check table no longer exists
+        check_table_doesnt_exist(integration, table_name)
+
+    finally:
+        for flow_id in flow_ids_to_delete:
+            delete_flow(client, flow_id)
+
+
+# Checking the successful deletion case and unsuccessful deletion case works as expected.
+# To test this, I have two workflows that write to the same table. When I delete the table in the first workflow,
+# it is successful but when I delete it in the second workflow, it is unsuccessful because the table has already
+# been deleted.
```

Comment:
Thanks for the comments, super useful! I've been putting them as multi-line comments (""" """ format) inside the test functions, which I think is what the python style guide recommends.